src/lib.rs

```diff
@@ -128,6 +128,37 @@ use embedded_graphics::{
     transform::Transform,
 };
 
+/// Trait to convert unsigned into signed integer.
+trait SaturatingCast<T> {
```

Comment:
@rfuest I'd love to reuse SaturatingCast without having to copy it. Do you think it might be worth to extract it to somewhere?

Reply:
Maybe we could add it to `embedded-graphics-core`. Could you open an issue for this?

Reply:
I can (in a few), but I'm also thinking about creating a crate just for this. Saturating casts might be generally useful enough for that, I think?

Reply:
Sure, this could be useful for other applications. I'm too lazy to do this my self at the moment, but if you want to create a crate for this it would be appreciated.

It would also be useful to implement the opposite conversion cast from signed to unsigned, which I could have used some time ago.

Reply:
There's also the az crate that does this. I'll check it more deeply if I'll have the time, maybe we can go with that

Reply:
The `az` crate looks promising. Thanks for digging that up, I had searched previously but didn't find any suitable crate.

Reply:
I've thought about this a bit. `az` doesn't have `saturating_cast_neg`, which means we either have to contribute the operation to the crate (IMO it is a bit more ad-hoc than to "just" copy and paste into `az`), or we have to transform our usages of `saturating_cast_neg` into `saturating_cast` somehow.

Neither of these is a good way forward.

On the other hand, I also think e-g's `SaturatingCast` wouldn't fit neatly into `core` - it's not strictly graphics related, and it's also rather special.

So for now I think I'll either keep the copy, or use `az` in e-t, but I'm not sure I'd touch e-g's implementation for this problem at this time.

Reply:
The missing `saturating_cast_neg` isn't a huge problem IMO. If we change the `value.saturating_cast_neg()` calls to `-value.saturating_cast()` the behavior will be almost identical. I don't think that saturating at `i32::min_value() + 1` instead of `i32::min_value()` would be a huge issue.